Update to Netbeans 20

mebenn opened the pull request:
3 plugins avaialble for C/C++ support in NB:
   - CPPLite Kit (with C/C++ Language server, but in early stages of development)
   - CND branch: uses CPPLite with some (much?) of the old CND merged (have not really tested this)
   - the old CND Plugin from 8.2 Plugin portal (this can be installed with this PR)
The last version of NB that supports the 8.2 CND plugin is 20 (but I think 21 should do as well). This is because NB has switched Java versions in 20/21.
While the C/C++ Language Server is the way to go for the future, currently the old CND 8.2 Plugin provides the ususal frills.

I usually use another editor to develop and I haven't really tested this to any significant degree of detail except a few projects.

This version was originally released in December 2023.

All the latest NB plugins should work with this version.

To port NB 25 to OI would, I suspect, require some effort.

mebenn committed:
Update to Netbeans 20

AndWac commented:
This doesn't build here. It may be related to the fact that my default java version is 21:
    [java] > startup failed:
     [java]   General error during conversion: Unsupported class file major version 65
     [java]
     [java]   java.lang.IllegalArgumentException: Unsupported class file major version 65
     [java]   	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:199)
     [java]   	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:180)
     [java]   	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:166)
     [java]   	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:287)
     [java]   	at org.codehaus.groovy.ast.decompiled.AsmDecompiler.parseClass(AsmDecompiler.java:81)
     [java]   	at org.codehaus.groovy.control.ClassNodeResolver.findDecompiled(ClassNodeResolver.java:251)
     [java]   	at org.codehaus.groovy.control.ClassNodeResolver.tryAsLoaderClassOrScript(ClassNodeResolver.java:189)
     [java]   	at org.codehaus.groovy.control.ClassNodeResolver.findClassNode(ClassNodeResolver.java:169)
     [java]   	at org.codehaus.groovy.control.ClassNodeResolver.resolveName(ClassNodeResolver.java:125)
     [java]   	at org.codehaus.groovy.ast.decompiled.AsmReferenceResolver.resolveClassNullable(AsmReferenceResolver.java:57)
     [java]   	at org.codehaus.groovy.ast.decompiled.AsmReferenceResolver.resolveClass(AsmReferenceResolver.java:44)
     [java]   	at org.codehaus.groovy.ast.decompiled.TypeSignatureParser.visitEnd(TypeSignatureParser.java:110)
     [java]   	at groovyjarjarasm.asm.signature.SignatureReader.parseType(SignatureReader.java:206)
     [java]   	at groovyjarjarasm.asm.signature.SignatureReader.accept(SignatureReader.java:124)
     [java]   	at org.codehaus.groovy.ast.decompiled.ClassSignatureParser.parseClassSignature(ClassSignatureParser.java:74)
     [java]   	at org.codehaus.groovy.ast.decompiled.ClassSignatureParser.configureClass(ClassSignatureParser.java:32)
     [java]   	at org.codehaus.groovy.ast.decompiled.DecompiledClassNode.lazyInitSupers(DecompiledClassNode.java:185)
     [java]   	at org.codehaus.groovy.ast.decompiled.DecompiledClassNode.isUsingGenerics(DecompiledClassNode.java:86)
     [java]   	at org.codehaus.groovy.ast.tools.GenericsUtils.nonGeneric(GenericsUtils.java:275)
     [java]   	at org.codehaus.groovy.ast.decompiled.MemberSignatureParser.createMethodNode(MemberSignatureParser.java:101)
     [java]   	at org.codehaus.groovy.ast.decompiled.DecompiledClassNode.lambda$createMethodNode$1(DecompiledClassNode.java:230)
     [java]   	at org.codehaus.groovy.ast.decompiled.DecompiledClassNode.createMethodNode(DecompiledClassNode.java:236)
     [java]   	at org.codehaus.groovy.ast.decompiled.DecompiledClassNode.lazyInitMembers(DecompiledClassNode.java:203)
     [java]   	at org.codehaus.groovy.ast.decompiled.DecompiledClassNode.getDeclaredMethods(DecompiledClassNode.java:122)
     [java]   	at org.codehaus.groovy.ast.ClassNode.tryFindPossibleMethod(ClassNode.java:1283)
     [java]   	at org.codehaus.groovy.control.StaticImportVisitor.transformMethodCallExpression(StaticImportVisitor.java:251)
     [java]   	at org.codehaus.groovy.control.StaticImportVisitor.transform(StaticImportVisitor.java:133)
     [java]   	at org.codehaus.groovy.ast.ClassCodeExpressionTransformer.visitExpressionStatement(ClassCodeExpressionTransformer.java:108)
     [java]   	at org.codehaus.groovy.ast.stmt.ExpressionStatement.visit(ExpressionStatement.java:40)
     [java]   	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitClassCodeContainer(ClassCodeVisitorSupport.java:138)
     [java]   	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitConstructorOrMethod(ClassCodeVisitorSupport.java:111)
     [java]   	at org.codehaus.groovy.ast.ClassCodeExpressionTransformer.visitConstructorOrMethod(ClassCodeExpressionTransformer.java:66)
     [java]   	at org.codehaus.groovy.control.StaticImportVisitor.visitConstructorOrMethod(StaticImportVisitor.java:108)
     [java]   	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitConstructor(ClassCodeVisitorSupport.java:101)
     [java]   	at org.codehaus.groovy.ast.ClassNode.visitContents(ClassNode.java:1089)
     [java]   	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitClass(ClassCodeVisitorSupport.java:52)
     [java]   	at org.codehaus.groovy.control.CompilationUnit.lambda$addPhaseOperations$3(CompilationUnit.java:209)
     [java]   	at org.codehaus.groovy.control.CompilationUnit$IPrimaryClassNodeOperation.doPhaseOperation(CompilationUnit.java:942)
     [java]   	at org.codehaus.groovy.control.CompilationUnit.processPhaseOperations(CompilationUnit.java:671)
     [java]   	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:635)
     [java]   	at groovy.lang.GroovyClassLoader.doParseClass(GroovyClassLoader.java:389)
     [java]   	at groovy.lang.GroovyClassLoader.lambda$parseClass$3(GroovyClassLoader.java:332)
     [java]   	at org.codehaus.groovy.runtime.memoize.StampedCommonCache.compute(StampedCommonCache.java:163)
     [java]   	at org.codehaus.groovy.runtime.memoize.StampedCommonCache.getAndPut(StampedCommonCache.java:154)
     [java]   	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:330)
     [java]   	at org.gradle.groovy.scripts.internal.DefaultScriptCompilationHandler.compileScript(DefaultScriptCompilationHandler.java:139)
     [java]   	at org.gradle.groovy.scripts.internal.DefaultScriptCompilationHandler.compileToDir(DefaultScriptCompilationHandler.java:95)
     [java]   	at org.gradle.groovy.scripts.internal.BuildOperationBackedScriptCompilationHandler$2.run(BuildOperationBackedScriptCompilationHandler.java:54)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:68)
     [java]   	at org.gradle.groovy.scripts.internal.BuildOperationBackedScriptCompilationHandler.compileToDir(BuildOperationBackedScriptCompilationHandler.java:51)
     [java]   	at org.gradle.groovy.scripts.internal.FileCacheBackedScriptClassCompiler$CompileToCrossBuildCacheAction.execute(FileCacheBackedScriptClassCompiler.java:190)
     [java]   	at org.gradle.groovy.scripts.internal.FileCacheBackedScriptClassCompiler$CompileToCrossBuildCacheAction.execute(FileCacheBackedScriptClassCompiler.java:170)
     [java]   	at org.gradle.groovy.scripts.internal.FileCacheBackedScriptClassCompiler$ProgressReportingInitializer.execute(FileCacheBackedScriptClassCompiler.java:211)
     [java]   	at org.gradle.groovy.scripts.internal.FileCacheBackedScriptClassCompiler$ProgressReportingInitializer.execute(FileCacheBackedScriptClassCompiler.java:194)
     [java]   	at org.gradle.cache.internal.DefaultPersistentDirectoryCache$Initializer.initialize(DefaultPersistentDirectoryCache.java:100)
     [java]   	at org.gradle.cache.internal.FixedSharedModeCrossProcessCacheAccess$1.run(FixedSharedModeCrossProcessCacheAccess.java:86)
     [java]   	at org.gradle.cache.internal.DefaultFileLockManager$DefaultFileLock.doWriteAction(DefaultFileLockManager.java:216)
     [java]   	at org.gradle.cache.internal.DefaultFileLockManager$DefaultFileLock.writeFile(DefaultFileLockManager.java:206)
     [java]   	at org.gradle.cache.internal.FixedSharedModeCrossProcessCacheAccess.open(FixedSharedModeCrossProcessCacheAccess.java:83)
     [java]   	at org.gradle.cache.internal.DefaultCacheAccess.open(DefaultCacheAccess.java:139)
     [java]   	at org.gradle.cache.internal.DefaultPersistentDirectoryStore.open(DefaultPersistentDirectoryStore.java:89)
     [java]   	at org.gradle.cache.internal.DefaultPersistentDirectoryStore.open(DefaultPersistentDirectoryStore.java:43)
     [java]   	at org.gradle.cache.internal.DefaultCacheFactory.doOpen(DefaultCacheFactory.java:103)
     [java]   	at org.gradle.cache.internal.DefaultCacheFactory.open(DefaultCacheFactory.java:68)
     [java]   	at org.gradle.cache.internal.DefaultCacheRepository$PersistentCacheBuilder.open(DefaultCacheRepository.java:117)
     [java]   	at org.gradle.groovy.scripts.internal.FileCacheBackedScriptClassCompiler.compile(FileCacheBackedScriptClassCompiler.java:116)
     [java]   	at org.gradle.groovy.scripts.internal.CrossBuildInMemoryCachingScriptClassCache.getOrCompile(CrossBuildInMemoryCachingScriptClassCache.java:50)
     [java]   	at org.gradle.groovy.scripts.internal.BuildScopeInMemoryCachingScriptClassCompiler.compile(BuildScopeInMemoryCachingScriptClassCompiler.java:50)
     [java]   	at org.gradle.groovy.scripts.DefaultScriptCompilerFactory$ScriptCompilerImpl.compile(DefaultScriptCompilerFactory.java:49)
     [java]   	at org.gradle.configuration.DefaultScriptPluginFactory$ScriptPluginImpl.apply(DefaultScriptPluginFactory.java:125)
     [java]   	at org.gradle.configuration.BuildOperationScriptPlugin$1.run(BuildOperationScriptPlugin.java:65)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:68)
     [java]   	at org.gradle.configuration.BuildOperationScriptPlugin.lambda$apply$0(BuildOperationScriptPlugin.java:62)
     [java]   	at org.gradle.configuration.internal.DefaultUserCodeApplicationContext.apply(DefaultUserCodeApplicationContext.java:44)
     [java]   	at org.gradle.configuration.BuildOperationScriptPlugin.apply(BuildOperationScriptPlugin.java:62)
     [java]   	at org.gradle.initialization.ScriptEvaluatingSettingsProcessor.applySettingsScript(ScriptEvaluatingSettingsProcessor.java:73)
     [java]   	at org.gradle.initialization.ScriptEvaluatingSettingsProcessor.process(ScriptEvaluatingSettingsProcessor.java:66)
     [java]   	at org.gradle.initialization.SettingsEvaluatedCallbackFiringSettingsProcessor.process(SettingsEvaluatedCallbackFiringSettingsProcessor.java:34)
     [java]   	at org.gradle.initialization.RootBuildCacheControllerSettingsProcessor.process(RootBuildCacheControllerSettingsProcessor.java:47)
     [java]   	at org.gradle.initialization.BuildOperationSettingsProcessor$2.call(BuildOperationSettingsProcessor.java:50)
     [java]   	at org.gradle.initialization.BuildOperationSettingsProcessor$2.call(BuildOperationSettingsProcessor.java:47)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
     [java]   	at org.gradle.initialization.BuildOperationSettingsProcessor.process(BuildOperationSettingsProcessor.java:47)
     [java]   	at org.gradle.initialization.DefaultSettingsLoader.findSettingsAndLoadIfAppropriate(DefaultSettingsLoader.java:136)
     [java]   	at org.gradle.initialization.DefaultSettingsLoader.findAndLoadSettings(DefaultSettingsLoader.java:62)
     [java]   	at org.gradle.initialization.SettingsAttachingSettingsLoader.findAndLoadSettings(SettingsAttachingSettingsLoader.java:34)
     [java]   	at org.gradle.internal.composite.CommandLineIncludedBuildSettingsLoader.findAndLoadSettings(CommandLineIncludedBuildSettingsLoader.java:34)
     [java]   	at org.gradle.internal.composite.ChildBuildRegisteringSettingsLoader.findAndLoadSettings(ChildBuildRegisteringSettingsLoader.java:48)
     [java]   	at org.gradle.internal.composite.CompositeBuildSettingsLoader.findAndLoadSettings(CompositeBuildSettingsLoader.java:35)
     [java]   	at org.gradle.initialization.InitScriptHandlingSettingsLoader.findAndLoadSettings(InitScriptHandlingSettingsLoader.java:34)
     [java]   	at org.gradle.initialization.GradlePropertiesHandlingSettingsLoader.findAndLoadSettings(GradlePropertiesHandlingSettingsLoader.java:39)
     [java]   	at org.gradle.initialization.DefaultSettingsPreparer.prepareSettings(DefaultSettingsPreparer.java:31)
     [java]   	at org.gradle.initialization.BuildOperationFiringSettingsPreparer$LoadBuild.doLoadBuild(BuildOperationFiringSettingsPreparer.java:62)
     [java]   	at org.gradle.initialization.BuildOperationFiringSettingsPreparer$LoadBuild.run(BuildOperationFiringSettingsPreparer.java:57)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:68)
     [java]   	at org.gradle.initialization.BuildOperationFiringSettingsPreparer.prepareSettings(BuildOperationFiringSettingsPreparer.java:45)
     [java]   	at org.gradle.initialization.VintageBuildModelController.lambda$prepareSettings$2(VintageBuildModelController.java:85)
     [java]   	at org.gradle.internal.model.StateTransitionController.lambda$doTransition$12(StateTransitionController.java:227)
     [java]   	at org.gradle.internal.model.StateTransitionController.doTransition(StateTransitionController.java:238)
     [java]   	at org.gradle.internal.model.StateTransitionController.doTransition(StateTransitionController.java:226)
     [java]   	at org.gradle.internal.model.StateTransitionController.lambda$transitionIfNotPreviously$10(StateTransitionController.java:201)
     [java]   	at org.gradle.internal.work.DefaultSynchronizer.withLock(DefaultSynchronizer.java:34)
     [java]   	at org.gradle.internal.model.StateTransitionController.transitionIfNotPreviously(StateTransitionController.java:197)
     [java]   	at org.gradle.initialization.VintageBuildModelController.prepareSettings(VintageBuildModelController.java:85)
     [java]   	at org.gradle.initialization.VintageBuildModelController.prepareToScheduleTasks(VintageBuildModelController.java:70)
     [java]   	at org.gradle.internal.build.DefaultBuildLifecycleController.lambda$prepareToScheduleTasks$2(DefaultBuildLifecycleController.java:134)
     [java]   	at org.gradle.internal.model.StateTransitionController.lambda$doTransition$12(StateTransitionController.java:227)
     [java]   	at org.gradle.internal.model.StateTransitionController.doTransition(StateTransitionController.java:238)
     [java]   	at org.gradle.internal.model.StateTransitionController.doTransition(StateTransitionController.java:226)
     [java]   	at org.gradle.internal.model.StateTransitionController.lambda$maybeTransition$9(StateTransitionController.java:187)
     [java]   	at org.gradle.internal.work.DefaultSynchronizer.withLock(DefaultSynchronizer.java:34)
     [java]   	at org.gradle.internal.model.StateTransitionController.maybeTransition(StateTransitionController.java:183)
     [java]   	at org.gradle.internal.build.DefaultBuildLifecycleController.prepareToScheduleTasks(DefaultBuildLifecycleController.java:132)
     [java]   	at org.gradle.internal.buildtree.DefaultBuildTreeWorkPreparer.scheduleRequestedTasks(DefaultBuildTreeWorkPreparer.java:33)
     [java]   	at org.gradle.internal.buildtree.DefaultBuildTreeLifecycleController.lambda$doScheduleAndRunTasks$2(DefaultBuildTreeLifecycleController.java:89)
     [java]   	at org.gradle.composite.internal.DefaultIncludedBuildTaskGraph.withNewWorkGraph(DefaultIncludedBuildTaskGraph.java:75)
     [java]   	at org.gradle.internal.buildtree.DefaultBuildTreeLifecycleController.doScheduleAndRunTasks(DefaultBuildTreeLifecycleController.java:88)
     [java]   	at org.gradle.internal.buildtree.DefaultBuildTreeLifecycleController.lambda$runBuild$4(DefaultBuildTreeLifecycleController.java:106)
     [java]   	at org.gradle.internal.model.StateTransitionController.lambda$transition$6(StateTransitionController.java:166)
     [java]   	at org.gradle.internal.model.StateTransitionController.doTransition(StateTransitionController.java:238)
     [java]   	at org.gradle.internal.model.StateTransitionController.lambda$transition$7(StateTransitionController.java:166)
     [java]   	at org.gradle.internal.work.DefaultSynchronizer.withLock(DefaultSynchronizer.java:44)
     [java]   	at org.gradle.internal.model.StateTransitionController.transition(StateTransitionController.java:166)
     [java]   	at org.gradle.internal.buildtree.DefaultBuildTreeLifecycleController.runBuild(DefaultBuildTreeLifecycleController.java:103)
     [java]   	at org.gradle.internal.buildtree.DefaultBuildTreeLifecycleController.scheduleAndRunTasks(DefaultBuildTreeLifecycleController.java:69)
     [java]   	at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:31)
     [java]   	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
     [java]   	at org.gradle.internal.buildtree.ProblemReportingBuildActionRunner.run(ProblemReportingBuildActionRunner.java:49)
     [java]   	at org.gradle.launcher.exec.BuildOutcomeReportingBuildActionRunner.run(BuildOutcomeReportingBuildActionRunner.java:69)
     [java]   	at org.gradle.tooling.internal.provider.FileSystemWatchingBuildActionRunner.run(FileSystemWatchingBuildActionRunner.java:119)
     [java]   	at org.gradle.launcher.exec.BuildCompletionNotifyingBuildActionRunner.run(BuildCompletionNotifyingBuildActionRunner.java:41)
     [java]   	at org.gradle.launcher.exec.RootBuildLifecycleBuildActionExecutor.lambda$execute$0(RootBuildLifecycleBuildActionExecutor.java:40)
     [java]   	at org.gradle.composite.internal.DefaultRootBuildState.run(DefaultRootBuildState.java:128)
     [java]   	at org.gradle.launcher.exec.RootBuildLifecycleBuildActionExecutor.execute(RootBuildLifecycleBuildActionExecutor.java:40)
     [java]   	at org.gradle.internal.buildtree.DefaultBuildTreeContext.execute(DefaultBuildTreeContext.java:40)
     [java]   	at org.gradle.launcher.exec.BuildTreeLifecycleBuildActionExecutor.lambda$execute$0(BuildTreeLifecycleBuildActionExecutor.java:65)
     [java]   	at org.gradle.internal.buildtree.BuildTreeState.run(BuildTreeState.java:53)
     [java]   	at org.gradle.launcher.exec.BuildTreeLifecycleBuildActionExecutor.execute(BuildTreeLifecycleBuildActionExecutor.java:65)
     [java]   	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionExecutor$3.call(RunAsBuildOperationBuildActionExecutor.java:61)
     [java]   	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionExecutor$3.call(RunAsBuildOperationBuildActionExecutor.java:57)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
     [java]   	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
     [java]   	at org.gradle.launcher.exec.RunAsBuildOperationBuildActionExecutor.execute(RunAsBuildOperationBuildActionExecutor.java:57)
     [java]   	at org.gradle.launcher.exec.RunAsWorkerThreadBuildActionExecutor.lambda$execute$0(RunAsWorkerThreadBuildActionExecutor.java:36)
     [java]   	at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:270)
     [java]   	at org.gradle.internal.work.DefaultWorkerLeaseService.runAsWorkerThread(DefaultWorkerLeaseService.java:119)
     [java]   	at org.gradle.launcher.exec.RunAsWorkerThreadBuildActionExecutor.execute(RunAsWorkerThreadBuildActionExecutor.java:36)
     [java]   	at org.gradle.tooling.internal.provider.ContinuousBuildActionExecutor.execute(ContinuousBuildActionExecutor.java:103)
     [java]   	at org.gradle.tooling.internal.provider.SubscribableBuildActionExecutor.execute(SubscribableBuildActionExecutor.java:64)
     [java]   	at org.gradle.internal.session.DefaultBuildSessionContext.execute(DefaultBuildSessionContext.java:46)
     [java]   	at org.gradle.tooling.internal.provider.BuildSessionLifecycleBuildActionExecuter$ActionImpl.apply(BuildSessionLifecycleBuildActionExecuter.java:100)
     [java]   	at org.gradle.tooling.internal.provider.BuildSessionLifecycleBuildActionExecuter$ActionImpl.apply(BuildSessionLifecycleBuildActionExecuter.java:88)
     [java]   	at org.gradle.internal.session.BuildSessionState.run(BuildSessionState.java:69)
     [java]   	at org.gradle.tooling.internal.provider.BuildSessionLifecycleBuildActionExecuter.execute(BuildSessionLifecycleBuildActionExecuter.java:62)
     [java]   	at org.gradle.tooling.internal.provider.BuildSessionLifecycleBuildActionExecuter.execute(BuildSessionLifecycleBuildActionExecuter.java:41)
     [java]   	at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:63)
     [java]   	at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:31)
     [java]   	at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:58)
     [java]   	at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:42)
     [java]   	at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:47)
     [java]   	at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:31)
     [java]   	at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:65)
     [java]   	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
     [java]   	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
     [java]   	at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:39)
     [java]   	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
     [java]   	at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:29)
     [java]   	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
     [java]   	at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:35)
     [java]   	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
     [java]   	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.create(ForwardClientInput.java:78)
     [java]   	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.create(ForwardClientInput.java:75)
     [java]   	at org.gradle.util.internal.Swapper.swap(Swapper.java:38)
     [java]   	at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:75)
     [java]   	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
     [java]   	at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:55)
     [java]   	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
     [java]   	at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:63)
     [java]   	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
     [java]   	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
     [java]   	at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:84)
     [java]   	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:37)
     [java]   	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:104)
     [java]   	at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:52)
     [java]   	at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:297)
     [java]   	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
     [java]   	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
     [java]   	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
     [java]   	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
     [java]   	at java.base/java.lang.Thread.run(Thread.java:1583)
     [java]
     [java]   1 error
     [java]
     [java]
     [java] * Try:
     [java] > Run with --stacktrace option to get the stack trace.
     [java] > Run with --info or --debug option to get more log output.
     [java] > Run with --scan to get full insights.
     [java]
     [java] * Get more help at https://help.gradle.org
     [java]
     [java] BUILD FAILED in 3s
  [nbmerge] Failed to build target: all-gradle

mebenn commented:
Nope, it most certainly cannot build  with jdk21.  Any work-around possible here?

mebenn commented:
oops! i see you're trying to build NetBeans with Gradle!!! Don't think that will work either: Ant.

iigs commented:
I tested using your Makefile to build netbeans 25.  It build successfully with openjdk 21. I know nothing about netbeans for testing it.

These are some of the changes I made:

removed lines (not needed since that is the default now):
BUILD_BITS=64 # for binaries or 32_and_64 for libraries
BITS=64

changed:
COMPONENT_VERSION= 25
COMPONENT_ARCHIVE_HASH= sha256:a1828c8bbb2c2c50a9cd6daf9f918678fa7f003ef12b861bf1670a719e948183
COMPONENT_ARCHIVE_URL=  https://dlcdn.apache.org/$(COMPONENT_NAME)/$(COMPONENT_NAME)/$(COMPONENT_VERSION)/$(COMPONENT_ARCHIVE)

removed COMPONENT_SIG_URL= $(COMPONENT_ARCHIVE_URL).asc

I am skeptical this is needed:
REQUIRED_PACKAGES += developer/java/openjdk8

AndWac commented:
If openjdk8 is not being used during build the line is not only not necessary but wrong.
All of the REQUIRED_PACKAGES lines are manually added and have to be checked for new netbeans versions (and probably also for every build).

mebenn commented:
NB made a hugh transition between NB versions 21/22*: it moved from old (opnjdk 8 to 11 and is now at 17 for the build). The problem are the plugins, and, in particular, for this PR a decent C/C++ plugin; my real reason for targeting NB20. I suspect that anyone using C/C++ on NB is using this plugin and hence NB 20 (or older) or possibly NB 21  (but that's the next story).  

(I compiled NB 25-test Candidate a while back on OI, but not with my Makefile!), just to see what the problems might be; but I assumed that getting the Makefile to compile would have been a pain, so that is indeed good news. Thanks.

I know that installing ancient jdk versions on the build server is a major pain, but would we like to provide a C/C++ IDE on OI? (Maybe a question for the ML), because CPPLite  is not really usable, and the CND/CPPlite branch on NB-25 is still in some kind of limbo (as far as I can see---but I haven't looked deeply into this). Moreover starting NB-20 from the terminal brings the usual jdk warnings whatever platformn you are on. 

I intended (assuming I had time) to do NB25, and requesting that the NB pkg package for NB20 remain available (I know it is a big package and cleaning up the repo is important). Maybe there is an alternative pkg repo we could use for NB-20, at least for a while until   there is a usable C/C++ on newer NBs.

To the chase: 
anyone interested in a C/C++ on OI? then  we should try to provide NB-20; otherwise, oif nobooody is interested, we _only_ provide NB25 or later.
On NB-20 I've tested installing and using C/C++ and it works. Moreover most of the newer plugins should work (I tested the new Python plugin)

mebenn commented:
>If openjdk8 is not being used during build the line is not only not necessary but wrong.
>All of the REQUIRED_PACKAGES lines are manually added and have to be checked for new netbeans versions (and 
>probably also for every build).


You should use openjdk8 (possibly openjdk11, but I haven't tried this) to build: at a minimum you require this to load the C/C++ CND plugin (there is a work-around for jdk11 I think and poossibly even later jdk versions), but., openjdk8 is a runtime requirement for NB 20 to load the plugin, otherwise you're resorting to work-arounds. This will have to be changed, of necessity, for later NB versions, and then the C/C++ 8.2 Plugin Portal for the CND plugin will not work . The '8' in the Plugin portal is informative.
While you might be able to compile with openjdkX, but the (any) plugin might not load. Getting tests to work would be very helpful

mebenn commented:
Maybe, if we change the PR title from Netbeans-20 to something like  Netbeans-C/C++ (which really means Netbeans-20, 19, 18, ...; but NOT Netbeans-25). We've then got the advantage of some of the old plugins (8.2) portal  and most of the new plugins from Netbeans. 

As we have no (well insufficient) tests, it is really dangerous to mess around with Java Versions for the build and then to run. And before you select select NB-25 for release, you'll have to examine which plugins work. Nearly all new plugins work for NB-20. Don't get me wrong, it is great that  'iigs' compiled NB-25 using the Makefile, great; but some of the new plugins will probably not work, at least not yet... NB-25 is for a little later, maybe.

Is anyone interested in C/C++ in Netbeans on OI? If yes, then do we want to inlude using a very old openjdk version to build for NB-20? If the answer to either of these questions is 'no' then throw away  this PR. And move on to NB-24 or possibly NB-25.

Does this make some sense? 

Answer to another question:
For Netbeans, for each new version, there should alway be a consideration made as to which Java version to use to build and run.

iigs commented:
It is generally preferred to have the latest version of software.   Based on the following documentation, can the plugins that you want to run work using NetBeans 25? https://github.com/apache/netbeans-antora-tutorials/pull/19/commits/c720e08fb7558d127fcf11a7b922b40c12e62a2c

Since it appears that NetBeans 20 may not compile with openjdk 21, you may have to force the version of Java during compile.  That looks like it can be set adding JAVA_VERSION (i.e. JAVA_VERSION=8 ) after including common.mk -- you may want to look through make-rules/shared-macros.mk to see if anything else needs to be set.  

You do not want to have both REQUIRED_PACKAGES += developer/java/openjdk8 and REQUIRED_PACKAGES += runtime/java/openjdk21 in a package. . It is also better to make it clear which of the REQUIRED_PACKAGES are Build dependencies and which are runtime (have a runtime list and then have a Build dependencies list)

if you move forward with NetBeans 25, these were some additions I made to the netbeans.p5m file to get it to publish (the transform with the drop may need to changed if this is not appropriate):
<transform file path=usr/netbeans/25/ide/bin/nativeexecution/SunOS-x86_64/unbuffer.so -> default pkg.linted.userland.action001.2 true >
<transform path=.*/libflatlaf-linux.*$ -> drop>
(it should be noted that I did not use the netbeans.p5m provided in your PR--I just added the two lines to the netbeans.p5m that came from the netbeans 12.6 recipe)

mebenn commented:
Didn't get much time to follow this; but I think I know a bit more.  There are 2 things to get right: building Netbeans with Java (I think this is REQUIRED_PACKAGES? is this correct), then loading the all important C/C++ CND plugin; but that only uses something _to load_ the Plugin (I think, or am nearly sure). So that would ony mean a runtime dependency on jdk8 (there may be a  work-around like supplying the file as an extra thing; but that is less desirable).
I compiled the Netbeans-20 using jdk8 and I've used it on-and-off for a while now. But someone else, who originally asked me for this on OI, has been using it more intensively and is really happy--can't unfortunately contact this person at the moment. Now compiling NB-25 is possible  (thanks); but a lot of plugins are not passing tests, so maybe NB-24 would be better?  But my concern is C/C++ CND support  (I've just loaded it on NB-25, but I haven't really tested it, which I'd really like to do. At least it loads and that is very good)
I recall having compile problems with jdk21, so I moved to jdk17. I can't remember what these were. 
   
> It is generally preferred to have the latest version of software. Based on the following documentation, can the plugins that you want to run work using NetBeans 25? [apache/netbeans-antora-tutorials@c720e08](https://github.com/apache/netbeans-antora-tutorials/commit/c720e08fb7558d127fcf11a7b922b40c12e62a2c)
> 
> Since it appears that NetBeans 20 may not compile with openjdk 21, you may have to force the version of Java during compile. That looks like it can be set adding JAVA_VERSION (i.e. JAVA_VERSION=8 ) after including common.mk -- you may want to look through make-rules/shared-macros.mk to see if anything else needs to be set.
> 
> You do not want to have both REQUIRED_PACKAGES += developer/java/openjdk8 and REQUIRED_PACKAGES += runtime/java/openjdk21 in a package. . It is also better to make it clear which of the REQUIRED_PACKAGES are Build dependencies and which are runtime (have a runtime list and then have a Build dependencies list)
 
Didn't really realise at the time, but i will be careful 
> 
> if you move forward with NetBeans 25, these were some additions I made to the netbeans.p5m file to get it to publish (the transform with the drop may need to changed if this is not appropriate): <transform file path=usr/netbeans/25/ide/bin/nativeexecution/SunOS-x86_64/unbuffer.so -> default pkg.linted.userland.action001.2 true > <transform path=._/libflatlaf-linux._$ -> drop> (it should be noted that I did not use the netbeans.p5m provided in your PR--I just added the two lines to the netbeans.p5m that came from the netbeans 12.6 recipe)

I'm not too well versed in transforms, it  was quicker to write a quick script to remove all the libs/exe/Linux/Mac... from the manifest, ... and checked the diffs that only appropriate  stuff was removed

If nobody is interested in a C/C++ IDE on OI; then the way forward is easy and no need to waste anymore time (most users will want the IDE for Java development). Anyway the old CND C/C++ isn't really supported on NB, but there a a lot of people using it.

What is the best way forward?  I'm really not sure, I'm definitely open to suggestions

hoewweken commented:
Hello!

We have Code::Blocks 20.3 and QTcreator 7.0.2 maybe you want to have a look.
I also would be interested what ide the OI/illumos developers are actually using (?) 

Regards,
Stephan

iigs commented:
> Didn't get much time to follow this; but I think I know a bit more. There are 2 things to get right: building Netbeans with Java (I think this is REQUIRED_PACKAGES? is this correct), then loading the all important C/C++ CND plugin; but that only uses 

mtelka says one does not to worry about having the right version of jdk specified in REQUIRED_PACKAGES since he says other parts of the .mk files should automatically detect that.  My advice with this regard is if you know the jdk is not installed and the package doesn't stop early on to tell the user to install the package, then the jdk package needs to be added in the Build Dependency section.

> something _to load_ the Plugin (I think, or am nearly sure). So that would ony mean a runtime dependency on jdk8 (there may be a work-around like supplying the file as an extra thing; but that is less desirable). I compiled the Netbeans-20 using jdk8 and I've used it on-and-off for a while now. But someone else, who originally asked me for this on OI, has been using it more intensively and is really happy--can't unfortunately contact this person at the moment. Now compiling NB-25 is possible (thanks); but a lot of plugins are not passing tests, so maybe NB-24 would be better? But my concern is C/C++ CND support (I've just loaded it on NB-25, but I haven't really tested it, which I'd really like to do. At least it loads and that is very good) I recall having compile problems with jdk21, so I moved to jdk17. I can't remember what these were.

The default java for builds is 21 in OI userland right now.  You need to specify JAVA_VERSION if you want to build with a different version of java.  Please look at  the openjdk support roadmap to determine which java version to choose based on the expected lifetime of the package: https://access.redhat.com/articles/1299013 .

OpenIndina will consider to drop openjdk 8 once security updates are no longer available (this may get delayed if there is a critical system component which still needs it).  It should also be noted that different openjdk contributors have given different dates when openjdk 8 is expected to cease getting updates. One needs to plan for any other openjdk release to go away once it is not actively getting security updates.

Would you consider being the maintainer for NetBeans for OpenIndiana and make sure there is a working version for your use case and a working version with the latest release (a new package would need to be added and tested every three months or so)? If so, this is how to consider things:
You can see what other distros have for their NetBeans versions:
https://repology.org/project/netbeans/versions
You can see the list of CVEs:
https://repology.org/project/netbeans/versions
Versions of Netbeans with CVEs would be expected to be obsoleted.
Older versions of NetBeans would be expected to be obsoleted.
I don't have a recommendation what the details should be -- this is needs to come from somebody more familiar with NetBeans.

It is recommended to make sure that OpenIndiana gets any versions with a CVE updated.  It is recommended that the versions of Netbeans use a supported version of Java.

The build recipe that you wrote will need to be tweaked to have the version number in the COMPONENT_NAME, i.,e. netbeans-20, netbeans-25, etc
In netbeans.p5m , the files/links for in /usr/bin and /usr/share would have to be as mediator.
It is ok if you want to manually delete files such as macos, linux, etc from the .p5m file.  You will eventually find it would be easier to maintain with transforms.  It took me years to get comfortable with transforms.

> 
> > It is generally preferred to have the latest version of software. Based on the following documentation, can the plugins that you want to run work using NetBeans 25? [apache/netbeans-antora-tutorials@c720e08](https://github.com/apache/netbeans-antora-tutorials/commit/c720e08fb7558d127fcf11a7b922b40c12e62a2c)
> > Since it appears that NetBeans 20 may not compile with openjdk 21, you may have to force the version of Java during compile. That looks like it can be set adding JAVA_VERSION (i.e. JAVA_VERSION=8 ) after including common.mk -- you may want to look through make-rules/shared-macros.mk to see if anything else needs to be set.
> > You do not want to have both REQUIRED_PACKAGES += developer/java/openjdk8 and REQUIRED_PACKAGES += runtime/java/openjdk21 in a package. . It is also better to make it clear which of the REQUIRED_PACKAGES are Build dependencies and which are runtime (have a runtime list and then have a Build dependencies list)
> 
> Didn't really realise at the time, but i will be careful
> 
> > if you move forward with NetBeans 25, these were some additions I made to the netbeans.p5m file to get it to publish (the transform with the drop may need to changed if this is not appropriate): <transform file path=usr/netbeans/25/ide/bin/nativeexecution/SunOS-x86_64/unbuffer.so -> default pkg.linted.userland.action001.2 true > <transform path=._/libflatlaf-linux._$ -> drop> (it should be noted that I did not use the netbeans.p5m provided in your PR--I just added the two lines to the netbeans.p5m that came from the netbeans 12.6 recipe)
> 
> I'm not too well versed in transforms, it was quicker to write a quick script to remove all the libs/exe/Linux/Mac... from the manifest, ... and checked the diffs that only appropriate stuff was removed
> 
> If nobody is interested in a C/C++ IDE on OI; then the way forward is easy and no need to waste anymore time (most users will want the IDE for Java development). Anyway the old CND C/C++ isn't really supported on NB, but there a a lot of people using it.
> 
> What is the best way forward? I'm really not sure, I'm definitely open to suggestions

I think it is great that you want to develop software on OpenIndiana. For those of us who are familiar with NetBeans, assistance is needed to make sure there is a workable solution for users.  OpenIndiana is project for users maintained by users.

iigs commented:
The github editor ate some characters.  There should be an asterisk after the period.  This seems to gets translated to remove the asterisk and enable/disable italics in markdown.
 
> <transform path=.*/libflatlaf-linux.*$ -> drop>

mebenn commented:
In answer to some questions:
  _Would you consider being the maintainer for NetBeans for OpenIndiana ..._
I'm afraid not. I'm particularly unsuited to maintaing Netbeans. I'm not really Java, so if there is a new Java version bump, I'd be ill-suited to update; apart from the time often devoured by Java bumps.  Moreover, I don't even use Netbeans; except ocasionally  for C/C++ or sometimes Python (the new Python language support (language server of course) is coming along nicesly and very usable.) Most people using NetBeans would be there for Java. Morever it is a very nice IDE.
For day-to-day use I live in another editor and I won't wean away from that too hastily.
Besides, there are a few other things I would like to complete: cppcheck. This is a C and C++ static analyzer (another? well, you can't have enough). Finally, there's embarisingly a PR lurking about for a couple of years: Glances. It is a kind-of htop but with some nice features. I should finish that before even thinking about committing to other things.  
That said, I'll always be there to answer questions and perform a package update time permitting, etc. 
Wouldn't it be nice to package Netbeans Release Candidates which could be tested on OI!  

_We have Code::Blocks 20.3 and QTcreator 7.0.2 maybe you want to have a look._
Yes, I know of Code::Blocks and delighted to see it on OI; but as I said above, I've got another abode for development.
'QTcreator' sounds like Qt libs, which I don't believe are available on OI; so it is probably not available on OI.

_The default java for builds is 21 in OI userland right now. You need to specify JAVA_VERSION .._
Usually JAVA_VERSION is sufficient to use the specified Java version.  I suspect ant-build has its own mind as to this.
It probably means scouring the endless build.xml files scattered about the place; but hopefully not. 
I'm simply glad to have been able to compile it with _any_ Java AND run the C/C++; but I think there I've figured that out. To install CND C/C++, remove the new CppLite; then load the CND C/C++ plugin which might fail if you are using a newer Java version, i.e., unpack200. However, the volks at Netbeans now have a very helpful message and instructions in how ot easily fix this: even those very same volks who are flustered into having to support ancient software. 
The merged CND beanch I coould get to work satisfactorily. 

 _The github editor ate some characters. There should be an asterisk after the period. This seems to gets translated to remove the asterisk and enable/disable italics in markdown._
 _<transform path=./libflatlaf-linux.$ -> drop>_
I hope the Manifest ascii file is ok in an editor. Its only a GitHub visual thing, I hope. 

_I think it is great that you want to develop software on OpenIndiana. ..._
and have been for quite some time. I  do occasionaly use NetBeans.

AndWac commented:
This has been superceded by #21303
